modules/healthcheck/default.nix

```
@@ -0,0 +1,281 @@
+                          }
+
+                          if [[ -n "''${NOTIFY_SOCKET:-}" ]]; then
+                            check &
+                          else
+                            echo "Health check: NOTIFY_SOCKET not set. Cannot run readiness probe." >&2
+                          fi
+
+                          ${cfg.exec}
+                        ''
+                      );
+                  }
+                ))
+              ) servicesWithHealthcheck)
+              // lib.mapAttrs' (
+                mainServiceName: serviceConfig:
+                let
+                  cfg = serviceConfig.healthcheck;
+                in
+                {
+                  name = "${mainServiceName}-liveness-check";
+                  value = mkIf (cfg != null && cfg.liveness-probe.enable) (
+                    let
+                      probe = cfg.liveness-probe;
+                      checkScript = pkgs.writeShellScriptBin "liveness-check" ''
+                        #!${pkgs.runtimeShell}
+                        sleep ${toString probe.initialDelaySeconds}
+                        echo "Executing liveness probe for ${mainServiceName}..."
+                        # If the command fails, explicitly restart the main service
+                        while true; do
+                          if ! (timeout ${toString probe.timeoutSeconds}s ${probe.command}); then
+                            echo "(timeout ${toString probe.timeoutSeconds}s ${probe.command})"
+                            echo "Liveness probe for ${mainServiceName} failed. Triggering restart..."
+                            ${pkgs.systemd}/bin/systemctl restart ${mainServiceName}.service &
+                          fi
+                          sleep ${toString probe.periodSeconds}
+                        done
+                      '';
+                    in
+                    {
+                      description = "Liveness check for ${mainServiceName}";
+                      # This check needs systemctl in its path.
+                      path = [ pkgs.systemd ] ++ (if cfg ? "path" then cfg.path else [ ]);
+                      serviceConfig = {
+                        Type = "oneshot";
+                        ExecStart = "${checkScript}/bin/liveness-check";
+                      };
+                    }
+                  );
+                }
+              ) servicesWithHealthcheck;
+          };
+        };
+
+      options.mcl.services = mkOption {
+        default = { };
+        type =
+          with types;
+          attrsOf (
+            submodule (
+              { name, config, ... }:
+              {
+                options = {
+                  healthcheck = mkOption {
+                    type =
+                      with types;
+                      nullOr (submodule {
+                        options = {
+                          # Programs to add to the PATH for the health check.
+                          path = mkOption {
```

Maybe rename this for clarity (not to be confused with a filesystem path or something):

```suggestion
                          runtimePackages = mkOption {
```